tips/TIP-0018/tip-0018.md

```diff
@@ -0,0 +1,3767 @@
+    - The new dust protection mechanism does not need a distinct output type, therefore
+      <i>SigLockedDustAllowanceOutput</i> will be deprecated. One alternative is that during migration to the new
+      protocol version, all dust outputs sitting on an address will be merged into an <i>ExtendedOutput</i> together
+      with their respective <i>SigLockedDustAllowanceOutputs</i> to create the snapshot for the updated protocol.
+      The exact migration strategy will be decided later.
+- Adding new [output types](#output-design) to _Transaction Payload_.
+- Adding new [unlock block types](#unlocking-chain-script-locked-outputs) to _Transaction Payload_.
+- <i>Inputs</i> and <i>Outputs</i> of a transaction become a list instead of a set. Binary duplicate inputs are not
+  allowed as they anyway mean double-spends, but binary duplicate outputs are allowed.
+- There can be many outputs created to the same address in the transaction.
+- Confirming milestone supplies notion of time to semantic transaction validation.
+
+## New Concepts
+
+New output types add new features to the protocol and hence new transaction validation rules. While some of these new
+features are specifically tied to one output type, some are general, LEGO like building blocks that may be put in
+several types of outputs.
+
+Below is a summary of such  new features and the validation rules they introduce.
+
+### Native Tokens in Outputs
+
+Outputs are records in the UTXO ledger that track ownership of funds. Thus, each output must be able to specify which
+funds it holds. With the addition of the Native Tokenization Framework, outputs may also carry user defined native
+tokens, that is, tokens that are not IOTA coins but were minted by foundries and are tracked in the very same ledger.
+Therefore, **every output must be able to hold not only IOTA coins, but also native tokens**.
+
+Dust protection applies to all outputs, therefore it is not possible for outputs to hold only native tokens, the dust
+requirements must be covered via IOTA coins.
+
+User defined tokens are called <i>Native Tokens</i> on protocol level. The maximum supply of a particular native token
+is defined by the representation chosen on protocol level for defining their amounts in outputs. Since native tokens
+are also a vehicle to wrap layer 2 tokens into layer 1 tokens, the chosen representation must take into account the
+maximum possible supply of layer 2 tokens. Solidity, the most popular layer 2 smart contract language defines the
+maximum supply of an ERC-20 token as `MaxUint256`, therefore it should be possible to represent such huge amount of
+assets on layer 1.
+
+Outputs must have the following fields to define the balance of native tokens they hold:
+
+<table>
+    <tr>
+        <td><b>Name</b></td>
+        <td><b>Type</b></td>
+        <td><b>Description</b></td>
+    </tr>
+    <tr>
+        <td>Native Tokens Count</td>
+        <td>uint8</td>
```

Comment:
I know there is a discussion about limits but as a note: the current maximum number of native tokens of 256 doesn't fit in an u8 anymore it should at least be decreased to 255